symfit/core/minimizers.py

```diff
@@ -64,8 +67,21 @@ class GradientMinimizer(BaseMinimizer):
     def __init__(self, *args, **kwargs):
         jacobian = kwargs.pop('jacobian')
         super(GradientMinimizer, self).__init__(*args, **kwargs)
-        self.jacobian = jacobian
-
+        def _jacobian(*args, **kwargs):
+            # Assume the order of what jacobian returns and parameters is the same
+            # We need to remove the values for the fixed parameters from what
+            # jacobian returns
+            fixed_vals = {p.name: p.value for p in self._fixed_params}
+            out = jacobian(*args, **kwargs, **fixed_vals)
+            jac = []
+            for param, val in zip(self._parameters, out):
```

Comment:
since you forced jacobian to be a numpy array (why?) it's cleaner to use numpy masking:
```python
mask = [p.fixed for p in self.parameters]
jac = out[mask]
```

Reply:
It's only forced to np.array in the scipy minimizer wrapper. Otherwise the scipy minimizer craps out.